### PR TITLE
API breakage allowlist

### PR DESCRIPTION
**Pull request by OdNairy**
PR https://github.com/mapbox/mapbox-maps-ios/pull/1095 introduced legit breaking changes that we would like to ignore. This pull request adds functionality to allow some breaks from the filelist. 

## Pull request checklist:
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add documentation comments for any added or updated public APIs.
 - [ ] Add any new public, top-level symbols to the Jazzy config's `custom_categories` (scripts/doc-generation/.jazzy.yaml)
 - [ ] Describe the changes in this PR, especially public API changes.
 - [ ] Add a changelog entry to to bottom of the relevant section (typically the `## main` heading near the top).
 - [ ] Update the guides (internal access only), README.md, and DEVELOPING.md if their contents are impacted by these changes.
 - [x] Review and agree to the Contributor License Agreement ([CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement)).

### Useful links:
- https://github.com/apple/swift/blob/main/test/api-digester/breakage-allowlist.swift

### CODE REVIEW

File: scripts/api-compatibility-check/.baseapi [hunk 1]
```diff
@@ -1 +1 @@
-81618929724d94fcc64d4be24caed15357e98446
+v10.3.0
```
Comment (macdrevx) [inline scripts/api-compatibility-check/.baseapi:1]: This was lower before, right? 10.1 or something like that?

**Review by macdrevx — COMMENTED** [review]
Question about the baseline

File: scripts/api-compatibility-check/.baseapi (hunk 1 above)
Comment (OdNairy) [reply scripts/api-compatibility-check/.baseapi:1]: We have to use the latest public release every time. 
I would like to automate this part this week.

File: scripts/api-compatibility-check/breakage_allowlist.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// https://github.com/mapbox/mapbox-maps-ios/pull/1095
+Func MapView.didMoveToSuperview() has been removed
+Func MapView.willMove(toWindow:) has been removed
+
+// _spi(Experimental)
+Enum PinchGestureBehavior has been removed
+Var GestureOptions.pinchBehavior has been removed
```
Comment (OdNairy) [inline scripts/api-compatibility-check/breakage_allowlist.txt:7]: @macdrevx I'd like to name each breaking change in this file (and remove them just after the new public release). What is the best name for these functions, WDYT?

**Review by macdrevx — APPROVED** [review]

**Pull request merged**